dashboard/app/models/course.rb

```diff
@@ -121,4 +121,57 @@ def summarize
       end
     }
   end
+
+  @@course_cache = nil
+  COURSE_CACHE_KEY = 'course-cache'.freeze
+
+  def self.should_cache?
+    return false if Rails.application.config.levelbuilder_mode
+    ENV['UNIT_TEST'] || ENV['CI']
+  end
+
+  # generates our course_cache from what is in the Rails cache
+  def self.course_cache_from_cache
+    Course.connection
```

Comment:
I do not know. I copied it because it was there in the script version. My assumption was that it perhaps somehow locked the db connection for this course of this method?